cpu/rpx0xx/periph/spi.c

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2023 Frank Engelhardt
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_rpx0xx
+ * @{
+ *
+ * @file
+ * @brief       Implementation of SPI.
+ *
+ * @author      Frank Engelhardt <frank@f9e.de>
+ *
+ * @}
+ */
+
+#include "assert.h"
+#include "bitarithm.h"
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief   Allocate one lock per SPI device.
+ */
+static mutex_t locks[SPI_NUMOF];
+
+/**
+ * @brief   Save the clock prescaler values for faster bus acquisition.
+ */
+typedef struct {
+    spi_clk_t clk;
+    uint8_t cpsdvsr;
+    uint8_t scr;
+} _pl022_clk_t;
+static _pl022_clk_t pl022_clk[SPI_NUMOF];
+
+gpio_t spi_pin_clk(spi_t spi)
+{
+    assert((unsigned)spi < SPI_NUMOF);
+    return spi_config[spi].clk_pin;
+}
+
+gpio_t spi_pin_miso(spi_t spi)
+{
+    assert((unsigned)spi < SPI_NUMOF);
+    return spi_config[spi].miso_pin;
+}
+
+gpio_t spi_pin_mosi(spi_t spi)
+{
+    assert((unsigned)spi < SPI_NUMOF);
+    return spi_config[spi].mosi_pin;
+}
+
+static void _poweron(spi_t spi)
+{
+    uint32_t reset_bit_mask =
+        (spi) ? RESETS_RESET_spi1_Msk : RESETS_RESET_spi0_Msk;
```

Comment:
Shouldn't that condition be `spi_config[spi].dev == SPI0`?

```suggestion
    uint32_t reset_bit_mask = spi_config[spi].dev == SPI0
                            ? RESETS_RESET_spi0_Msk
                            : RESETS_RESET_spi1_Msk;
```